packages/esm-patient-forms-app/src/forms/forms-see-all.tsx

```diff
@@ -25,13 +25,8 @@ enum FormViewState {
 
 const Forms = () => {
   const urlPathArray = window.location.pathname.split('/');
-  const patientUuid = urlPathArray[urlPathArray.length - 2];
+  const patientUuid = urlPathArray[urlPathArray.length - 3];
```

Comment:
@manuelroemer Could you please tell me in which file this extension slot is defined? I am sorry, this part of the framework is a little unclear to me.

Reply:
One slot lives, for example, in `patient-chart-app` (you can look into `chart-review.component` and follow the code there). But the important part here is that the extension (i.e. the `Forms` component here) is provided with a `patientUuid` prop by the slot(s). This means that you can simply do this in the component:

```ts
interface FormsProps {
  patientUuid: string
}

const Forms: React.FC<FormsProps> = ({ patientUuid }) => { ... }
```

And as long as _that_ component is used as the extension, you will receive the current `patientUuid` in there which means that you don't have to extract it yourself.

An example for this is e.g. in `esm-patient-immunizations-app`. There are two extensions there:

```ts
extensions: [
  {
    id: 'immunization-overview-widget',
    slot: 'patient-chart-summary-dashboard-slot',
    load: getAsyncLifecycle(() => import('./immunizations/immunizations-overview.component'), options),
    meta: {
      columnSpan: 2,
    },
  },
  {
    id: 'immunization-details-widget',
    slot: dashboardMeta.slot,
    load: getAsyncLifecycle(() => import('./immunizations/immunizations-detailed-summary.component'), options),
    meta: {
      columnSpan: 1,
    },
  },
```
And if you follow the two components (`./immunizations/immunizations-overview.component` and `./immunizations/immunizations-detailed-summary.component`) you can see that they simply have a `patientUuid` prop which is automatically provided.

Reply:
Thanks, that makes things clearer and I think along with patientUuid even a fhir.Patient object gets passed as a prop to the extension. Both of these props are accesible to the extensions. But what if I want to pass a different prop to the extension as suggested by you:
#321 (comment)

How can a prop of this type (Eg. pageSize) be passed to an extension from the slot and which file do I need to access to do this for the forms extension?

Reply:
Yes, absolutely, using that `patient` is even better than refetching the data yourself with the `patientUuid`.

About the prop from the other component: I'd keep it simple and not mess with the slot at all. It shouldn't be the slot's responsibility to know how the `pageSize` of your component anyway. -> This means that the slot _won't_ pass the prop to your extension.

What I meant with my other comment is the following:

```ts
// Modify the Forms component so that the pageSize is now a prop.
interface FormsProps {
  patient: Patient;
  pageSize: number;
}

const Forms: React.FC<FormsProps> = ({ patient, pageSize }) => {
  // Your component as it is right now, only that the pageSize is now a prop
}
```

And then create two wrapper components **which are used as the extensions**:

```ts
// The Forms component to be rendered in the summary (with 5 rows): 
const FormsDashboard = ({ patient }) => {
  return <Forms pageSize={5} patient={patient} />;
};
```

```ts
// The "See All" component (with 10 rows):
const FormsSeeAll = ({ patient }) => {
  return <Forms pageSize={10} patient={patient} />;
};
```

This way, the two wrappers don't need a `pageSize` prop from the slot, but _can_ receive the `patient` from it and pass it down to your actual `Forms` component. (I didn't add types in the above examples, but I hope that the general idea is clear).

Reply:
Thanks for the detailed explanation. I have implented this in the second commit.